hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+        set interfaces {port} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port=port, vlan_id=vlan_id)
+
+        self._set_commit_config(add_vlan)
+
+    def _remove_vlan_from_port(self, port, vlan_id):
+        """ removes a single vlan specified by `vlan_id` """
+        remove_vlan = """
+        delete interfaces {port} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port=port, vlan_id=vlan_id)
+        self._set_commit_config(remove_vlan)
+        
+    def _remove_all_vlans_port(self, port):
+        """ Removes all vlans from the port, including the native vlan.
+        Also converts the interface to access mode. 
+        """
+        remove_all_vlans = """
+            delete interfaces {port} unit 0 family ethernet-switching vlan
+            delete interfaces {port} unit 0 family ethernet-switching interface-mode
+            delete interfaces {port} native-vlan-id
+            """.format(port=port)
+        self._set_commit_config(remove_all_vlans)
+
+    def _get_port_networks(self, port):
+        """ Returns list of all the vlans shared with the port. """
+        port_info = self._interface_info(port)
+        return port_info[port]['vlans'] 
+
+    def _set_port_state(self, port, state):
+        """ Disables or enables the `port`. 
+        Depending on the value of `state`.
+        """
+
+        if state == "enable":
+            set_port_state = """
+            delete interface {port} disable
+            """.format(port=port)
+        elif state == "disable":
+            set_port_state = """
+            set interface {port} disable
+            """.format(port=port)
+        self._set_commit_config(remove_vlan)
+
+    def revert_port(self, port):
+        """Resets the port to the factory default.
+        Running this function will replace the port settings
+        with the ones stored in the jinja template `jinja_templates/`
+        For a physical port, it will use the template `revert_nonchannelized_port.j2`
+        For a channelized port ( 4 10G ports made from 1 40G port) it will use
+        template `revert_channelized_port.j2`
+        """
+        port_name = { 'port_name': port }
+        jun = self.session()
+        rm_curr_port_config = """
+        delete interfaces {iface}
+        """.format(iface=port_name['port_name'])
+
+        if port[-2] == ":":
+            base_config='junos/jinja_templates/revert_channelized_port.j2'
+        else:
+            base_config='junos/jinja_templates/revert_nonchannelized_port.j2'
+        
+        try:
+            with jun:
+                jun.cfg.load(rm_curr_port_config, format="set")
+                jun.cfg.load( 
+                        template_path = base_config, template_vars=port_name,
+                        format='text'
+                        )
+
+                if jun.cfg.commit_check:
+                    jun.cfg.commit
+                else:
+                    jun.cfg.rollback
```

Comment:
We can call `_commit_config` here in place of this if-else block

Reply:
I agree. replaced it as suggested.